NuGetProtocolSamples/Program.cs

```diff
@@ -0,0 +1,141 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Packaging;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace NuGet.Protocol.Samples
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            Console.WriteLine("Listing package versions...");
+            await ListPackageVersionsAsync();
+
+            Console.WriteLine();
+            Console.WriteLine("Downloading package...");
+            await DownloadPackageAsync();
+
+            Console.WriteLine();
+            Console.WriteLine("Get package metadata...");
+            await GetPackageMetadataAsync();
+
+            Console.WriteLine();
+            Console.WriteLine("Search packages..");
+            await SearchPackages();
+        }
+
+        public static async Task ListPackageVersionsAsync()
+        {
+#region ListPackageVersions
```

Comment:
This region is used by the docs website: https://github.com/NuGet/docs.microsoft.com-nuget/pull/1889/files#diff-e64dec9f9a43116a20fea96fd2557aa9R33

Reply:
The only appropriate use of regions, lol :D 
Should we add a comment above the regions?

Reply:
Added!